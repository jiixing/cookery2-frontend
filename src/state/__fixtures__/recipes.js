const recipes = () => ({
  entities: {
    recipes: {
      1: {
        id: 1,
        name: 'test name',
        servings: 2,
        difficulty: 'easy',
        duration: 0,
      },
      2: {
        id: 2,
        name: 'Bell’s Expedition',
        servings: 5,
        difficulty: 'easy',
        duration: 0,
      },
      3: {
        id: 3,
        name: 'Founders Kentucky Breakfast ',
        servings: 2,
        difficulty: 'medium',
        duration: 0,
      },
      4: {
        id: 4,
        name: 'Chimay Grande Réserve',
        servings: 3,
        difficulty: 'medium',
        duration: 0,
      },
      5: {
        id: 5,
        name: 'Alpha King Pale Ale',
        servings: 1,
        difficulty: 'difficult',
        duration: 0,
      },
      6: {
        id: 6,
        name: 'La Fin Du Monde',
        servings: 6,
        difficulty: 'easy',
        duration: 0,
      },
      7: {
        id: 7,
        name: 'Trappistes Rochefort 10',
        servings: 1,
        difficulty: 'difficult',
        duration: 0,
      },
      10: {
        id: 10,
        name: 'Yeti Imperial Stout',
        servings: 2,
        difficulty: 'medium',
        duration: 0,
      },
      11: {
        id: 11,
        name: 'Samuel Smith’s Oatmeal Stout',
        servings: 6,
        difficulty: 'medium',
        duration: 93,
      },
      15: {
        id: 15,
        name: 'my new thing',
        servings: 3,
        difficulty: 'easy',
        duration: 0,
      },
      16: {
        id: 16,
        name: 'name name name',
        servings: 3,
        difficulty: 'medium',
        duration: 0,
      },
    },
  },
  result: [1, 2, 3, 4, 5, 6, 7, 10, 11, 15, 16],
})

export default recipes
