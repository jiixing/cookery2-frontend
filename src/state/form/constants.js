export const namespace = 'form'
export const recipeModel = 'recipe'

export const RECIPE_CREATE_SUCCESS = 'recipes:recipe:create:success'
export const RECIPE_CREATE_FAIL = 'recipes:recipe:create:fail'
