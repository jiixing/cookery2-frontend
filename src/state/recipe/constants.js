export const namespace = 'recipe'

export const REQUEST_RECIPE = `${namespace}:request`
export const RECEIVE_RECIPE = `${namespace}:receive`
