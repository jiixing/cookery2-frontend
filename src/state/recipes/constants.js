export const namespace = 'recipes'

export const REQUEST_RECIPES = `${namespace}:request`
export const RECEIVE_RECIPES = `${namespace}:receive`
