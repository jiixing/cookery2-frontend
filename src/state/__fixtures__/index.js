import recipes from './recipes'

const state = { recipes: recipes() }

export { recipes }
export default () => state
