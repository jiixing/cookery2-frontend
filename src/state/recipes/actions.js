// @flow

import { fetch, normalizeRecipes } from '../../rest'

import { REQUEST_RECIPES, RECEIVE_RECIPES } from './constants'

import type { RecipeSummary } from '../../types'

const requestRecipes = () => ({
  type: REQUEST_RECIPES
})

const receiveRecipes = (recipes: Array<RecipeSummary>) => ({
  type: RECEIVE_RECIPES,
  payload: recipes
})

export function fetchRecipes () {
  return (dispatch: Dispatch) => {
    dispatch(requestRecipes())

    return fetch('/recipes')
      .then(response => response.json())
      .then(json => normalizeRecipes(json))
      .then(recipes => dispatch(receiveRecipes(recipes)))
  }
}
