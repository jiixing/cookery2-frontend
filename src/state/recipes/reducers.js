import { REQUEST_RECIPES, RECEIVE_RECIPES } from './constants'

const initialState = {
  entities: {
    recipes: {},
  },
}

function recipes(state = initialState, action) {
  switch (action.type) {
    case REQUEST_RECIPES:
      return initialState
    case RECEIVE_RECIPES:
      return action.payload
    default:
      return state
  }
}

export default recipes
