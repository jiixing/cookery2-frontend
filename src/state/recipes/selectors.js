import * as R from 'ramda'

import { namespace } from './constants'

const recipesSelector = R.path([namespace, 'entities', 'recipes'])
export default recipesSelector
