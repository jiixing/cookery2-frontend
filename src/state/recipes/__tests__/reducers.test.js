import { recipes as recipesState } from '../../__fixtures__'

import { REQUEST_RECIPES, RECEIVE_RECIPES } from '../constants'
import recipes from '../reducers'

describe('recipes reducer', () => {
  it('returns the initial state', () => {
    expect(recipes(undefined, {})).toMatchSnapshot()
  })

  it('handles REQUEST_RECIPES', () => {
    const action = {
      type: REQUEST_RECIPES
    }

    expect(recipes(recipesState(), action)).toMatchSnapshot()
  })

  it('handles RECEIVE_RECIPES', () => {
    const action = {
      type: RECEIVE_RECIPES,
      payload: recipesState()
    }

    expect(recipes(undefined, action)).toMatchSnapshot()
  })
})
