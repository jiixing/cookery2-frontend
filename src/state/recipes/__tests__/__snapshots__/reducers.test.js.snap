// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipes reducer handles RECEIVE_RECIPES 1`] = `
Object {
  "entities": Object {
    "recipes": Object {
      "1": Object {
        "difficulty": "easy",
        "duration": 0,
        "id": 1,
        "name": "test name",
        "servings": 2,
      },
      "10": Object {
        "difficulty": "medium",
        "duration": 0,
        "id": 10,
        "name": "Yeti Imperial Stout",
        "servings": 2,
      },
      "11": Object {
        "difficulty": "medium",
        "duration": 93,
        "id": 11,
        "name": "Samuel Smith’s Oatmeal Stout",
        "servings": 6,
      },
      "15": Object {
        "difficulty": "easy",
        "duration": 0,
        "id": 15,
        "name": "my new thing",
        "servings": 3,
      },
      "16": Object {
        "difficulty": "medium",
        "duration": 0,
        "id": 16,
        "name": "name name name",
        "servings": 3,
      },
      "2": Object {
        "difficulty": "easy",
        "duration": 0,
        "id": 2,
        "name": "Bell’s Expedition",
        "servings": 5,
      },
      "3": Object {
        "difficulty": "medium",
        "duration": 0,
        "id": 3,
        "name": "Founders Kentucky Breakfast ",
        "servings": 2,
      },
      "4": Object {
        "difficulty": "medium",
        "duration": 0,
        "id": 4,
        "name": "Chimay Grande Réserve",
        "servings": 3,
      },
      "5": Object {
        "difficulty": "difficult",
        "duration": 0,
        "id": 5,
        "name": "Alpha King Pale Ale",
        "servings": 1,
      },
      "6": Object {
        "difficulty": "easy",
        "duration": 0,
        "id": 6,
        "name": "La Fin Du Monde",
        "servings": 6,
      },
      "7": Object {
        "difficulty": "difficult",
        "duration": 0,
        "id": 7,
        "name": "Trappistes Rochefort 10",
        "servings": 1,
      },
    },
  },
  "result": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    10,
    11,
    15,
    16,
  ],
}
`;

exports[`recipes reducer handles REQUEST_RECIPES 1`] = `
Object {
  "entities": Object {
    "recipes": Object {},
  },
}
`;

exports[`recipes reducer returns the initial state 1`] = `
Object {
  "entities": Object {
    "recipes": Object {},
  },
}
`;
